Implemented routes

briefgaming opened the pull request:
get all products => returns all products listed in the db
update product => updates a products in the database

briefgaming committed:
Merge pull request #1 from Curiouspaul1/master

Updated views added login auth decorator, first attempt at fixing Int…

briefgaming committed:
Merge branch 'master' of https://github.com/Curiouspaul1/Farmspeys

Merged Paul's requirement

briefgaming committed:
added getallproducts route

briefgaming committed:
Merge pull request #2 from Bashorun97/product-routes

added getallproducts route

briefgaming committed:
added updateproducts route

briefgaming committed:
added productID row in Products table

briefgaming committed:
Merge pull request #3 from Bashorun97/product-routes

Product routes

briefgaming committed:
Create README.md

briefgaming committed:
Update README.md

briefgaming committed:
Merge https://github.com/Curiouspaul1/Farmspeys

Merge from paul

briefgaming committed:
more updates

briefgaming committed:
Merge branch 'master' of https://github.com/Bashorun97/Farmspeys

briefgaming committed:
added productid schem

briefgaming committed:
getallproducts

briefgaming committed:
added updateproduct route

Curiouspaul1 commented on app/api/views.py:
```diff
@@ -150,14 +150,40 @@ def getspaces():
+    product.space = current_space
+    db.session.commit()
+    
+    return({'msg':f'Updated products successfully'}), 200
```
lol and this didnt throw a syntax error at you when you tested - abi you didn't even test it before sending your PR, should be:
```suggestion
    return make_response(jsonify({'msg':f'Updated products successfully'}), 200)
```

Curiouspaul1 commented on app/api/views.py:
```diff
@@ -150,14 +150,40 @@ def getspaces():
+    product.discount = data['discount']
+    product.date_created = d.datetime.utcnow()
+
+    current_space = current_user.space
```
The narrative is that every product upon creation must have a space its bound to by relationship with its model - therefore there's no need to update that at this point. What we could do at most is to cross-check that the user that's logged is in-fact the owner of this product

Curiouspaul1 commented on app/api/views.py:
```diff
@@ -150,14 +150,40 @@ def getspaces():
 
+@api.route('/getallproducts', methods=['GET'])
+@login_required
+def getallproducts(current_user):
```
This getallproducts is explicit to every product in our database and should be accessible by an admin only, otherwise ensure that you're doing this for a specific space : this would mean that you need to add a second argument to streamline your query. Perhaps you might want to do this on a new route such as : "/getproducts/<storeId>"
```suggestion
def getallproducts(current_user):
        if current_user.email == current_app.config['FLASK_ADMIN'] or current_user.role.has_permission(PERMISSION.ADMIN):
```

briefgaming replied:
Alright. I'll add the functionality for an admin and a user (this would return all products belonging to a user in the database and a separate route for all products belonging to particular space for a user)

Curiouspaul1 commented on app/api/views.py:
```diff
@@ -76,7 +76,7 @@ def register_user():
 def login():
```
where's my  login route and decorator,, chai!

Curiouspaul1 replied:
where's my login route and decorator ..chai!

Curiouspaul1 commented on app/api/views.py:
```diff
@@ -150,14 +150,40 @@ def getspaces():
+    product.images = data['images']
+    product.Instock = data['available_stock']
+    product.discount = data['discount']
+    product.date_created = d.datetime.utcnow()
```
You definitely can't and shouldn't do this...the dat_created is static...only appended once (and the values is assigned upon creation which isn't what this route does)

briefgaming replied:
Noted

Curiouspaul1 reviewed (commented):
please work on the mentioned issues

briefgaming committed:
get_all_products endpoint for admin only

briefgaming committed:
added more routes

briefgaming committed:
help resolve bugs. PLEASE DEBUG ONLY IN THIS TESTING BRANCH EVEN THE MASTER HAS A FEW BUGS SO BE CAREFUL. 1.The permissions of all users automatically becomes equal to that of a user that get promoted to a seller or admin. Somewhat like all users permissions are linked to one another. 2. Having issues with creating an admin user by checking the mail. I tried debugging by removing current_app.config in place of directly feeding it with the mail. I should have tampered with something only. I'm actively debugging too

briefgaming committed:
help resolve bugs. PLEASE DEBUG ONLY IN THIS TESTING BRANCH EVEN THE MASTER HAS A FEW BUGS SO BE CAREFUL. 1.The permissions of all users automatically becomes equal to that of a user that get promoted to a seller or admin. Somewhat like all users permissions are linked to one another. 2. Having issues with creating an admin user by checking the mail. I tried debugging by removing current_app.config in place of directly feeding it with the mail. I should have tampered with something only. I'm actively debugging too

briefgaming committed:
test

briefgaming committed:
added back route

briefgaming committed:
Fixed issues relating to giving users add permissions.
Added dummy logic for considering having a an admin-only decorator for admin-centric routes.
Added routes for making updates to a couple of models

briefgaming committed:
Merge branch 'master' into tests

briefgaming committed:
Removed stupid mistakes from typos

briefgaming committed:
Fix to correctly assigning ADMIN privelege

briefgaming committed:
FLASKY_ADMIN gets value form os.environ.getenv

briefgaming committed:
restored User's super function to previous state

Bashorun97 commented:
Please review

Curiouspaul1 commented:
Please don't push code with bugs in it ..i'll merge this for now and try and fix the issues locally.
